src/components/subscriptions/SubscriptionExpirationModal.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { Modal, MailtoLink } from '@edx/paragon';
+import Cookies from 'universal-cookie';
+
+import { SubscriptionContext } from './SubscriptionData';
+import {
+  SUBSCRIPTION_EXPIRATION_FIRST_THRESHOLD,
+  SUBSCRIPTION_EXPIRATION_SECOND_THRESHOLD,
+  SUBSCRIPTION_EXPIRATION_THIRD_THRESHOLD,
+  SEEN_SUBSCRIPTION_EXPIRATION_MODAL_COOKIE_PREFIX,
+} from './constants';
+
+import { formatTimestamp } from '../../utils';
+
+import Img from '../../components/Img';
+import edxLogo from '../../images/edx-logo.png';
+
+
+function SubscriptionExpirationModal({ enterpriseSlug }) {
+  const { details } = useContext(SubscriptionContext);
+  const { daysUntilExpiration, expirationDate } = details;
+
+  const renderTitle = () => (
+    <small>Renew your subscription</small>
+  );
+
+  const renderBody = () => (
+    <React.Fragment>
+      <p>
+        Your current subscription is set to expire in {daysUntilExpiration} days.
+        In order to minimize course access disruptions for your learners, make sure your invoice has
+        been completed.
+      </p>
+      <p>
+        If you have questions or need help, please contact the edX Customer Success team at
+        {' '}
+        <MailtoLink to="customersuccess@edx.org">customersuccess@edx.org</MailtoLink>.
+      </p>
+      <i>
+        Access expires on {formatTimestamp({ timestamp: expirationDate })}
+      </i>
+    </React.Fragment>
+  );
+
+  const renderExpiredBody = () => (
+    <React.Fragment>
+      <Img className="w-25 my-5 mx-auto d-block" src={edxLogo} alt="edX logo" />
+      <p>
+        Your subscription license expired on {formatTimestamp({ timestamp: expirationDate })}.
+        To access your subscription management page contact edX and reactivate your subscriptions.
+      </p>
+      <p>
+        What to do next?
+      </p>
+      <ul>
+        <li>
+          To reactivate your subscriptions please cotact the edX Customer Success team at
+          {' '}
+          <MailtoLink to="customersuccess@edx.org">customersuccess@edx.org</MailtoLink>
+        </li>
+        <li>
+          View your learner progress in the <Link to={`/${enterpriseSlug}/admin/learners`}>learner management page</Link>
+        </li>
+        <li>
+          Manage your codes in the <Link to={`/${enterpriseSlug}/admin/coupons`}>code management page</Link>
+        </li>
+      </ul>
+    </React.Fragment>
+  );
+
+  // If the subscription has already expired, we show a different un-dismissible modal
+  const subscriptionExpired = daysUntilExpiration <= 0;
+  if (subscriptionExpired) {
+    return (
+      <Modal
+        renderHeaderCloseButton={false}
+        renderDefaultCloseButton={false}
+        title={null}
```

Comment:
Paragon doesn't really support no title, but BCool said let's go with this for now and he'll mention updating the modal to not require a title to the paragon working group soon

Reply:
Updated the styles for the post expiration header. Personally I'm not sold on the event listener being worth it since (as you pointed out):
- No real subscriptions are set to expire this calendar year
- We have to add in backend validation anyway
- The existing modal is getting replaced in Paragon "soon"

I could be convinced if you/others feel really strongly, but I feel like this is ok for now